sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClient.java

```diff
@@ -438,4 +443,147 @@ public IterableStream<ChatCompletions> getChatCompletionsStream(
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @return the {@link SyncPoller} for polling of status details for long running operations.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
+    public SyncPoller<BinaryData, BinaryData> beginStartGenerateImage(
+            BinaryData imageGenerationOptions, RequestOptions requestOptions) {
+        return this.serviceClient.beginStartGenerateImage(imageGenerationOptions, requestOptions);
+    }
+
+    /**
+     * Returns the status of the images operation.
+     *
+     * @param operationId .
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the result of the operation if the operation succeeded.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public ImageOperationResponse getImageOperationStatus(String operationId) {
+        // Generated convenience method for getImageOperationStatusWithResponse
+        RequestOptions requestOptions = new RequestOptions();
+        return getImageOperationStatusWithResponse(operationId, requestOptions)
+                .getValue()
+                .toObject(ImageOperationResponse.class);
+    }
+
+    /**
+     * Starts the generation of a batch of images from a text caption.
+     *
+     * @param imageGenerationOptions Represents the request data used to generate images.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws ClientAuthenticationException thrown if the request is rejected by server on status code 401.
+     * @throws ResourceNotFoundException thrown if the request is rejected by server on status code 404.
+     * @throws ResourceModifiedException thrown if the request is rejected by server on status code 409.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the {@link SyncPoller} for polling of status details for long running operations.
+     */
+    @Generated
+    @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
+    public SyncPoller<PollResult, ImageOperationStatus> beginStartGenerateImage(
```

Comment:
should we consider to rename this method name.

"beginStart" sounds the same thing.

Might be "beginGenerateImage" since it is LRO and should have started with `begin` prefix.

Reply:
This is a downstream change from the TSP definiton. Let's surface this feedback in the Wednesday meeting, if that works for you.

Reply:
@jpalvarezl 

There is something about the definition of this LRO in tsp that Java codegen being checking with author.
https://github.com/Azure/azure-rest-api-specs/pull/24382#discussion_r1236531864

And arch is checking on the terminal state.
https://github.com/Azure/azure-rest-api-specs/pull/24382/files#r1237679552

So, the LRO may not work as expected in runtime. If you do test it, please let me know the runtime log. Thanks in advance.

Reply:
Will do. I've made a work item around this. It might be a couple of days before we get to the actual unit testing of this. Will keep you posted.

Reply:
Srikanta did a test

The final result is

```json
{
  "created": 1687461794,
  "expires": 1687548201,
  "id": "bc67228d-12ec-45c7-b73a-3b8486465dfe",
  "result": {
    "created": 1687461794,
    "data": [
      {
        "url": "###"
      },
      {
...
      }
    ]
  },
  "status": "succeeded"
}
```

So the tsp definition on the LRO is wrong.

Reply:
typespec-java 0.8.3 released.

Fixed 2 issues:
- Do not generate API `getImageOperationStatus` (requested by Srikanta)
- Use `com.azure.core.experimental.util.polling.SyncOperationLocationPollingStrategy` in Impl

However the problem on the LRO in tsp source still exists. I would expect the 3rd temperate parameter be `ImageResponse`.

Hence the API in Java be
```java
    @Generated
    @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
    public SyncPoller<PollResult, ImageResponse> beginStartGenerateImage(
            ImageGenerationOptions imageGenerationOptions) {
        // Generated convenience method for beginStartGenerateImageWithModel
        RequestOptions requestOptions = new RequestOptions();
        return serviceClient.beginStartGenerateImageWithModel(
                BinaryData.fromObject(imageGenerationOptions), requestOptions);
    }
```

Impl be
```java
    @ServiceMethod(returns = ReturnType.LONG_RUNNING_OPERATION)
    public PollerFlux<PollResult, ImageResponse> beginStartGenerateImageWithModelAsync(
            BinaryData imageGenerationOptions, RequestOptions requestOptions) {
        return PollerFlux.create(
                Duration.ofSeconds(1),
                () -> this.startGenerateImageWithResponseAsync(imageGenerationOptions, requestOptions),
                new com.azure.core.experimental.util.polling.OperationLocationPollingStrategy<>(
                        new PollingStrategyOptions(this.getHttpPipeline())
                                .setEndpoint("{endpoint}/openai".replace("{endpoint}", this.getEndpoint()))
                                .setContext(
                                        requestOptions != null && requestOptions.getContext() != null
                                                ? requestOptions.getContext()
                                                : Context.NONE)),
                TypeReference.createInstance(PollResult.class),
                TypeReference.createInstance(ImageResponse.class));
    }
```

Reply:
Another topic: we also need some automation that runs when we update the autorest-java and typespec-java, to re-generate the SDK that has a CI built-in, so that the upgrade on codegen won't block the CI.

Reply:
As you can see in the sync scenario we never actually get to a `running` state. Could be an issue with the local setup. Here is the code for Async:

```java
public Mono<ImageOperationResponse> generateImage(ImageGenerationOptions imageGenerationOptions) {
  RequestOptions requestOptions = new RequestOptions();
  BinaryData imageGenerationOptionsBinaryData = BinaryData.fromObject(imageGenerationOptions);
  return serviceClient.beginStartGenerateImageAsync(imageGenerationOptions, requestOptions)
                .last()
                .flatMap(it -> it.getFinalResult())
                .map(it -> it.toObject(ImageOperationResponse.class));
}
```

And for the sync version. (the commented out `options` exhibit the same behaviour.

```java
RequestOptions requestOptions = new RequestOptions();
        return beginStartGenerateImage(BinaryData.fromObject(imageGenerationOptions), requestOptions)
// option 1
            .waitUntil(LongRunningOperationStatus.SUCCESSFULLY_COMPLETED).getValue()
// option 2
//            .getFinalResult()

// option 3
//            .waitForCompletion()
//            .getValue()
            .toObject(ImageOperationResponse.class);
```

Reply:
I wasn't aware that we had sync-stack enabled in openai. There were still a bug in pageable in current 1.40.0 core (that would be fixed in next 1.41.0 core).

Reply:
For the problem with sync polling, I am suspicious of a possible bug in `SimpleSyncPoller`. Investigating.

@samvaity saw one in purview as well. Maybe same cause.